x-pack/plugins/synthetics/public/hooks/use_date_format.ts

```diff
@@ -7,13 +7,18 @@
 
 import moment from 'moment';
 import { useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
 
 export function useDateFormat(): (timestamp?: string) => string {
-  const locale = navigator.language;
+  const kibanaLocale = i18n.getLocale();
+  const clientLocale = navigator.language;
 
   useEffect(() => {
-    moment.locale(locale);
-  }, [locale]);
+    const preferredLocale = kibanaLocale ?? clientLocale;
+    if (moment.locale() !== preferredLocale) {
+      moment.locale(preferredLocale);
+    }
```

Comment:
Since Kibana has it's own config level locale, it might not be a good idea to consider browser locale within a plugin.

I pushed this commit to prefer first the Kibana locale, and then the browser locale. Without this, there could be discrepancies between what locale is fed into Kibana vs. what locale the plugin is using. e.g. see below:

<img width="1202" alt="Screenshot 2023-07-04 at 18 34 42" src="https://github.com/elastic/kibana/assets/2748376/a41d81cc-61c9-457c-8487-f8bac287de0e">